src/main/scala/com/raphtory/core/model/graph/visitor/Vertex.scala

```diff
@@ -49,6 +51,38 @@ trait Vertex extends EntityVisitor {
   //individual in edge
   def explodeInEdge(id: Long,after:Long=0L,before:Long=Long.MaxValue): Option[List[ExplodedEdge]]
 
+  // weight
+  private def directedEdgeWeight(dir : Direction = EdgeDirection.Incoming, weightString : String="weight", mergeStrategy : Merge = MergeStrategy.Sum) : Float = {
+    val eWeights = ( dir match {
+      case EdgeDirection.Incoming =>
+        getInEdges()
+      case EdgeDirection.Outgoing =>
+        getOutEdges()
+      case EdgeDirection.Both =>
+        getEdges()
+    })
+      .map(e=>e.weightOrHistory(weightString))
+    mergeStrategy match {
+      case MergeStrategy.Sum => eWeights.sum
+      case MergeStrategy.Max => eWeights.max
+      case MergeStrategy.Min => eWeights.min
+      case MergeStrategy.Product => eWeights.product
+      case _ => 0.0f
```

Comment:
Same comment for the temporal vs spatial features. Also I would think if a strategy isn't specified than you don't return `0.0` but rather the default case (which I think in Raphtory is the latest updated weight).

Reply:
There should probably be no fall-through case here (i.e., specifying an unknown merge strategy should be an error). This is the reduction for combining edge weights, so latest doesn't make sense here.

Reply:
Have removed the fall-through case here, thanks!